### PR TITLE
Fix bug in `format_value` causing warning for NAs

### LINKED ISSUE #247: Warning in `formatters::format_value`: `"number of items to replace is not a multiple of replacement length"`
(transferred from `tern`)

An example occured in one of examples but the root-cause seems to be living here.

```
r$>  dta_test <- data.frame(
       Group = rep(LETTERS[1:3], each = 2),
       sub_group = rep(letters[1:2], each = 3),
       x = 1:6
     )

r$>  basic_table() %>%
       split_cols_by(var = "Group") %>%
       split_rows_by(var = "sub_group") %>%
       analyze(vars = "x", afun = s_summary) %>%
       build_table(df = dta_test)
                                                 A                        B                       C                  
—————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
a                                                                                                                    
  n                                              2                        1                       0                  
  sum                                            3                        3                       NA                 
  mean                                          1.5                       3                       NA                 
  sd                                     0.707106781186548               NA                       NA                 
  se                                            0.5                      NA                       NA                 
  mean_sd                              1.5, 0.707106781186548           3, NA                     NA                 
  mean_se                                     1.5, 0.5                  3, NA                     NA                 
  Mean 95% CI                   -4.85310236808735, 7.85310236808735      NA                       NA                 
  Mean -/+ 1xSE                                 1, 2                     NA                       NA                 
  Mean -/+ 1xSD                 0.792893218813452, 2.20710678118655      NA                       NA                 
  Mean p-value (H0: mean = 0)            0.204832764699133               NA                       NA                 
  median                                        1.5                       3                       NA                 
  mad                                            0                        0                       NA                 
  Median 95% CI                                  NA                      NA                       NA                 
  25% and 75%-ile                               1, 2                    3, 3                      NA                 
  iqr                                            1                        0                       NA                 
  range                                         1, 2                    3, 3                      NA                 
  min                                            1                        3                       NA                 
  max                                            2                        3                       NA                 
  Median (Min - Max)                         1.5, 1, 2                 3, 3, 3                    NA                 
  cv                                      47.1404520791032               NA                       NA                 
  geom_mean                               1.4142135623731                 3                       NA                 
  Geometric Mean 95% CI         0.0172997815631007, 115.608396135236     NA                       NA                 
  geom_cv                                 52.1092246837487               NA                       NA                 
b                                                                                                                    
  n                                              0                        1                       2                  
  sum                                            NA                       4                       11                 
  mean                                           NA                       4                      5.5                 
  sd                                             NA                      NA               0.707106781186548          
  se                                             NA                      NA                      0.5                 
  mean_sd                                        NA                     4, NA           5.5, 0.707106781186548       
  mean_se                                        NA                     4, NA                  5.5, 0.5              
  Mean 95% CI                                    NA                      NA      -0.853102368087346, 11.8531023680873
  Mean -/+ 1xSE                                  NA                      NA                      5, 6                
  Mean -/+ 1xSD                                  NA                      NA       4.79289321881345, 6.20710678118655 
  Mean p-value (H0: mean = 0)                    NA                      NA               0.057715876752609          
  median                                         NA                       4                      5.5                 
  mad                                            NA                       0                       0                  
  Median 95% CI                                  NA                      NA                       NA                 
  25% and 75%-ile                                NA                     4, 4                     5, 6                
  iqr                                            NA                       0                       1                  
  range                                          NA                     4, 4                     5, 6                
  min                                            NA                       4                       5                  
  max                                            NA                       4                       6                  
  Median (Min - Max)                             NA                    4, 4, 4                5.5, 5, 6              
  cv                                             NA                      NA                12.8564869306645          
  geom_mean                                      NA                       4                5.47722557505166          
  Geometric Mean 95% CI                          NA                      NA       1.71994304449266, 17.4424380482025 
  geom_cv                                        NA                      NA                12.945835316564           
There were 12 warnings (use warnings() to see them)

r$> warnings()
Warning messages:
1: In txt[is.na(txt)] <- na_str :
  number of items to replace is not a multiple of replacement length
2: In txt[is.na(txt)] <- na_str :
  number of items to replace is not a multiple of replacement length
3: In txt[is.na(txt)] <- na_str :
  number of items to replace is not a multiple of replacement length
4: In txt[is.na(txt)] <- na_str :
  number of items to replace is not a multiple of replacement length
5: In txt[is.na(txt)] <- na_str :
  number of items to replace is not a multiple of replacement length
6: In txt[is.na(txt)] <- na_str :
  number of items to replace is not a multiple of replacement length
7: In txt[is.na(txt)] <- na_str :
  number of items to replace is not a multiple of replacement length
8: In txt[is.na(txt)] <- na_str :
  number of items to replace is not a multiple of replacement length
9: In txt[is.na(txt)] <- na_str :
  number of items to replace is not a multiple of replacement length
10: In txt[is.na(txt)] <- na_str :
  number of items to replace is not a multiple of replacement length
11: In txt[is.na(txt)] <- na_str :
  number of items to replace is not a multiple of replacement length
12: In txt[is.na(txt)] <- na_str :
  number of items to replace is not a multiple of replacement length
```

Minimal reprex is as follows however not sure if this is a valid call:
```
r$> formatters::format_value(c(1, NA), format = "xx")
[1] "1"  "NA"
Warning message:
In txt[is.na(txt)] <- na_str :
  number of items to replace is not a multiple of replacement length
```

### PR DESCRIPTION
**Pull request by edelarua**
Closes #247

### CODE REVIEW

File: R/format_value.R [hunk 1]
```diff
@@ -300,8 +300,8 @@ format_value <- function(x, format = NULL, output = c("ascii", "html"), na_str =
         format, " are of different length"
       )
     }
-    if (length(na_str) < length(x)) {
-      na_str <- rep(na_str, length.out = length(x))
+    if (length(na_str) < sum(is.na(x))) {
```
Comment (Melkiades) [inline R/format_value.R:303]: Ok, then adding a vector is also possible. Do you know if this is documented? I am investigating

Comment (Melkiades) [reply R/format_value.R:303]: `format_value(c(1, NA, NA), "xx", na_str = c("as", "sa"))` This seems to be undocumented

Comment (Melkiades) [reply R/format_value.R:303]: Screenshot 2024-01-22 102433
Let's limit this. Can I ask you to merge the formatters `@param na_str` into one doc entry that specifies that you can insert a vector of the correct length? (I also suspect this has no tests or checks for the right match)

Comment (edelarua) [reply R/format_value.R:310]: @Melkiades I don't think the other two functions with `na_str` parameters need to be able to take vectors as `na_str` (unless we want to fully refactor `round_fmt` to accept vectors arguments for `digits`/`na_str`, but there doesn't seem to be a use case as of yet for this)

Comment (edelarua) [reply R/format_value.R:303]: @Melkiades on second thought, I think we should make a separate PR to tackle this issue - we have to also consider the fact that users can supply multiple `na_str` values for multi-value formats (e.g. `na_str = c("lo", "hi")` for `"xx.xx - xx.xx"`), which would complicate enabling vector input.

Comment (Melkiades) [reply R/format_value.R:303]: I see! Have you made it already? I think we need to understand how it is exactly applied and uniform that. In other words, we have to see exactly what providing a vector entails. From a user perspective, you would do 3 things: (1) one for all na_str (current default). (2) 2+ for specific formats (how do we select them externally? I do not remember, is na_str working as formats, that you can specify different values with named vectors?) (3) assign them all in order (if less, cycle)

**Review by Melkiades — APPROVED** [review]
lgtm!!! As soon as this is fixed we can group and decide `na_str` behavior ;) Thanks Emily for the fix

**Pull request merged**